pymodbus/client/serial.py

```diff
@@ -184,6 +183,8 @@ def __init__(
 
         self.last_frame_end = None
 
+        if not self.comm_params.baudrate:
```

Comment:
Actually it does not if you use "socket:". 

Apart from that if we catch baud rate, then why not stop bits....all these have a default value.

Reply:
Solved by https://github.com/pymodbus-dev/pymodbus/pull/1872